dbt_semantic_interfaces/validations/saved_query.py

```diff
@@ -0,0 +1,84 @@
+import logging
+import traceback
+from typing import Generic, List, Sequence, Set
+
+from dbt_semantic_interfaces.protocols import SemanticManifestT, SemanticModel
+from dbt_semantic_interfaces.protocols.saved_query import SavedQuery
+from dbt_semantic_interfaces.validations.validator_helpers import (
+    FileContext,
+    SavedQueryContext,
+    SavedQueryElementType,
+    SemanticManifestValidationRule,
+    ValidationError,
+    ValidationIssue,
+    generate_exception_issue,
+    validate_safely,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class SavedQueryRule(SemanticManifestValidationRule[SemanticManifestT], Generic[SemanticManifestT]):
+    """Validates fields in a saved query.
+
+    As the semantic model graph is not traversed completely in DSI, the validations for saved queries can't be complete.
+    Consequently, the current plan is that we add a separate validation using MetricFlow in CI.
+
+    * Check if metric names exist in the manifest.
+    * Check that the where filter is valid using the same logic as WhereFiltersAreParsable
```

Comment:
Do we also need to check `group_by_item_names` to check that they reference existing things? Additionally, what are valid `group_by_item_names`, is it just categorical dimensions?

Reply:
@QMalcolm Yeah, that would be good to check, but I didn't see similar checks being done for filters. Is that right?

Reply:
Oh right group_bys take the form of `Dimension(...)`. Then we need a validation similar to the `WhereFilter` validations that checks that they're parsable by the jinja templater, because as with `WhereFilters` the jinja template parser is the only thing can enforce the protocol structure of these structured strings.

Reply:
Changes to this look good!